packages/imperative/src/rest/src/session/ConnectionPropsForSessCfg.ts

```diff
@@ -307,7 +329,7 @@ export class ConnectionPropsForSessCfg {
             impLogger.debug("Using basic authentication");
             sessCfg.type = SessConstants.AUTH_TYPE_BASIC;
         }
-        ConnectionPropsForSessCfg.setTypeForTokenRequest(sessCfg, connOpts, cmdArgs.tokenType);
+        ConnectionPropsForSessCfg.setTypeForTokenRequest(sessCfg, connOpts as any, cmdArgs.tokenType);
```

Comment:
```suggestion
        ConnectionPropsForSessCfg.setTypeForTokenRequest<SessCfgType>(sessCfg, connOpts, cmdArgs.tokenType);
```

I think we can remove this `as any` here if we modify the signature of `setTypeForTokenRequest`

```typescript
    private static setTypeForTokenRequest<SessCfgType extends ISession=ISession>(
        sessCfg: SessCfgType,
        options: IOptionsForAddConnProps<SessCfgType>,
        tokenType: SessConstants.TOKEN_TYPE_CHOICES
    )
```